Source/Core/VideoCommon/NetPlayChatUI.h

```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include <deque>
+#include <functional>
+#include <memory>
+#include <string>
+#include <utility>
+
+class NetPlayChatUI
+{
+public:
+  explicit NetPlayChatUI(std::function<void(const std::string&)> callback);
+  ~NetPlayChatUI() = default;
+
+  void Display();
+  void AppendChat(const std::string& message, const std::string& color);
+  void SendMessage();
+
+private:
+  char m_message_buf[256] = {};
```

Comment:
This shouldn't need to be a member variable. Declare it right before ImGui::InputText and pass the result to SendMessage.

Reply:
This wouldn't retain the input value, so it would keep resetting every time it was drawn...

Reply:
Ah, didn't realize `ImGui::InputText` used it as it's state.